api/jobset/v1alpha2/jobset_types.go

```diff
@@ -261,8 +261,7 @@ type ReplicatedJob struct {
 	// If JobSet is suspended the all active ReplicatedJobs will be suspended. When JobSet is
 	// resumed the Job sequence starts again.
 	// This API is mutually exclusive with the StartupPolicy API.
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
-	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:Immutable
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
```

Please revert this. 

I have not seen this in the wild anywhere so I'm not sure this actually works. I also don't see anything added the CRD for JobSet so I am skeptical this actually does anything.

Reply:
If we remove this, we may need to specify the `MaxItems` to a fixed number to avoid error like: https://github.com/kubernetes-sigs/jobset/pull/878#discussion_r2049814155

WDYT about this? @andreyvelich @kannon92

Reply:
Can we just process with the same validation that we do for ReplicatedJob array: https://github.com/kubernetes-sigs/jobset/blob/379ec6d651709d09f14f16b934877b18d96f35dd/api/jobset/v1alpha2/jobset_types.go#L101 ?

Do we validate that this value must be immutable in the Webhook ?

Reply:
I am not convinced that Immutable tag actually does anything and I worry that this relaxes immutability.

Reply:
Yes, we haven't verified it, thus I asked whether we have any validation for other APIs where we use slices ?
For instance: `ReplicatedJobs []ReplicatedJob`

Reply:
Can we pls add an integration test?

Reply:
> Can we pls add an integration test?

Do you mean that we could add a integration test to verify that `+kubebuilder:validation:Immutable` works?

Reply:
yes

Reply:
I asked around (thank you @JoelSpeed!) and there is no mention of the Immutable field in controller-tools or the kubebuilder book.

https://book.kubebuilder.io/reference/markers/crd-validation#crd-validation

Reply:
@kannon92 WDYT about this? I think this could mitigate your worries:)

Reply:
Thanks for pointing this out. I refer to generative AI for advice on this, maybe it just give me the wrong answer:)

Reply:
So should we put a limit to the `MaxItems` or just removing this validation? @kannon92 @andreyvelich

Reply:
SGTM

Reply:
Sure, we can add limits as @ahg-g proposed.
Do we also want to consider to add limits for number of ReplicatedJobs in JobSet ?

Reply:
@ahg-g do you have a good feel for the impact of adding validation of 5 to existing users?

Adding validation is a breaking change so want to make sure we pick a value that is not going to impact a large amount of users.

Reply:
this is relaxing validation (from 1 to 5), so it is backward compatible, right?

Reply:
Yes, you are right.  Doh.